### PR TITLE
[Testing:Submission] Cypress Gradeable Test Case Tests

### PR DESCRIPTION
**Pull request by mahi-pas**
Partially Fixes Issue #8295 

### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant

### What is the new behavior?
Tests: 
- Playing confetti animation
- Showing details of an individual test case
- Hiding details of a loaded test case
- Canceling a loading test case (https://github.com/Submitty/Submitty/pull/7986)
- Expand/Collapse All Test Cases (https://github.com/Submitty/Submitty/pull/8217)

### CODE REVIEW

File: site/cypress/e2e/gradeables/gradeable_results.spec.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    beforeEach(() => {
+        cy.visit(['sample','gradeable','closed_team_homework']);
+        cy.login('student');
+
+    });
+
+    it('Should display confetti', () => {
+        cy.get('#confetti_canvas[style="display: block;"]').should('not.exist');
+        cy.get('div[onclick="addConfetti();"]').click();
+        cy.get('#confetti_canvas[style="display: block;"]').should('exist');
```
Comment (williamjallen) [inline site/cypress/e2e/gradeables/gradeable_results.spec.js:13]: These need to be targeted in a more specific way.  Targeting based on a JS function name or an inline CSS style is fragile.  It would be more robust to target these elements based on an ID or class, or via a data tag similar to what Cameron did in #9503.

File: site/cypress/e2e/gradeables/gradeable_results.spec.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    });
+
+    it('Should show and hide details', () => {
+        //Open
+        cy.get('#testcase_1').should('have.css', 'display').and('match', /none/);
+        cy.get('.loading-tools-show').eq(1).click();
+        cy.get('#testcase_1').should('have.css', 'display').and('not.match', /none/);
+        //Close
+        cy.get('.loading-tools-hide').first().click();
+        cy.get('#testcase_1').should('have.css', 'display').and('match', /none/);
```
Comment (williamjallen) [inline site/cypress/e2e/gradeables/gradeable_results.spec.js:23]: We should not be targeting things based on CSS properties for any reason.  It would be better to use `.should('exist')` or .should('be.visible')` here.  Apply throughout.

**Review by williamjallen — CHANGES_REQUESTED** [review]

**Review by williamjallen — CHANGES_REQUESTED** [review]
It looks like you've checked in a bunch of irrelevant changes here which need to be reverted.

**Review by 04vmatsibekker — CHANGES_REQUESTED** [review]
I believe you have a bad merge into main, and it is making the code look like a lot of files are changed. Please go through the code and fix the merge main mistake.

File: site/cypress/e2e/gradeables/gradeable_results.spec.js (hunk 1 above)
Comment (mahi-pas) [reply site/cypress/e2e/gradeables/gradeable_results.spec.js]: done

File: site/cypress/e2e/gradeables/gradeable_results.spec.js (hunk 2 above)
Comment (mahi-pas) [reply site/cypress/e2e/gradeables/gradeable_results.spec.js]: done

**Comment by mahi-pas** [comment]
Fixed the merge stuff, now it's down from 1,500 lines to just 50 lines added lol

File: site/cypress/e2e/gradeables/gradeable_results.spec.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+describe('Testing functionality of Autograder Results', () => {
+    beforeEach(() => {
+        cy.visit(['sample','gradeable','closed_team_homework']);
+        cy.login('student');
+
+    });
+
+    it('Should display confetti', () => {
+        cy.get('#confetti_canvas').should('have.css', 'display').and('not.match',/block/);
```
Comment (IDzyre) [inline site/cypress/e2e/gradeables/gradeable_results.spec.js:11]: I don't have a working VM to test this locally, is there a reason to test the CSS 'display' instead of using .should('be.visible') or .should('exist')  (Similar to William's previous comment)

Comment (williamjallen) [reply site/cypress/e2e/gradeables/gradeable_results.spec.js:11]: Yep, this needs to be fixed before this PR can be merged.

**Review by williamjallen — CHANGES_REQUESTED** [review]
@IDzyre's comment needs to be addressed before this can be merged.

File: site/cypress/e2e/gradeables/gradeable_results.spec.js (hunk 3 above)
Comment (mahi-pas) [reply site/cypress/e2e/gradeables/gradeable_results.spec.js:11]: The reason I didn't change it is because should('be.visible') doesn't work on confetti canvas because its ancestor element has position:fixed. The way it's displayed in js code is by changing the display property, so querying this specifically is better for testing the code.

**Comment by mahi-pas** [comment]
It wasn't passing because github CI doesn't simulate autograding, thus the tests would never find any proper values. I talked with Barb and she said we should leave it as a local only test. As for the testing css display value for confetti, the way the js changes the visibilty of the confetti is by changing the css display value, so it's the best way to test. 

TLDR:
- This test wont work on github CI, it should be kept as a local test
- CSS display value for confetti is the only way I can think of to test confetti
